[ENH] Relax the naming conventions of the `test_pkg_linkage` in v2 testing

**Describe the bug**
<!--
A clear and concise description of what the bug is.
-->
Currently, the `test_pkg_linkage` test in `test_all_estimators_v2.py` enforces a strict naming convention between:

- the estimator class (e.g. `MyForecaster_v2`)
- its corresponding package class (e.g. `MyForecaster_v2_pkg_v2`)

However, when migrating models from v1 to v2, this constraint becomes unnecessarily restrictive.

If a model already exists in v1 as:
```python
class MyForecaster:
```
and we implement its v2 version as:
```python
class MyForecaster_v2:
```
then the test forces the corresponding package class to be named: `MyForecaster_v2_pkg_v2`. This leads to awkward and redundant naming such as: `MyForecaster_v2_pkg_v2`.
In practice, a more natural name would be: `MyForecaster_pkg_v2`.
But this currently fails the test.

**To Reproduce**
<!--
Add a Minimal, Complete, and Verifiable example (for more details, see e.g. https://stackoverflow.com/help/mcve

If the code is too long, feel free to put it in a public gist and link it in the issue: https://gist.github.com
-->
The issue originates from this assertion:
```python
assert object_pkg.__name__ == object_class.__name__ + "_pkg_v2", msg
```

Look at this conversation: https://github.com/sktime/pytorch-forecasting/pull/2040#discussion_r2811980647
The error looks like this
```
object_pkg = <class 'pytorch_forecasting.models.xlstm._xlstm_pkg_v2.xLSTMTime_pkg_v2'>
object_class = <class 'pytorch_forecasting.models.xlstm._xlstm_v2.xLSTMTime_v2'>

    def test_pkg_linkage(self, object_pkg, object_class):
        """Test that the package is linked correctly."""
        # check name method
        msg = (
            f"Package {object_pkg}.name() does not match class "
            f"name {object_class.__name__}. "
            "The expected package name is "
            f"{object_class.__name__}_pkg."
        )
        assert object_pkg.name() == object_class.__name__, msg

        # check naming convention
        msg = (
            f"Package {object_pkg.__name__} does not match class "
            f"name {object_class.__name__}. "
            "The expected package name is "
            f"{object_class.__name__}_pkg."
        )
>       assert object_pkg.__name__ == object_class.__name__ + "_pkg_v2", msg
E       AssertionError: Package xLSTMTime_pkg_v2 does not match class name xLSTMTime_v2. The expected package name is xLSTMTime_v2_pkg.     
E       assert 'xLSTMTime_pkg_v2' == 'xLSTMTime_v2_pkg_v2'
E
E         - xLSTMTime_v2_pkg_v2
E         ?           ---
E         + xLSTMTime_pkg_v2

pytorch_forecasting\tests\test_all_v2\test_all_estimators_v2.py:134: AssertionError
```
**Suggested solution**
We should make some changes to this line https://github.com/sktime/pytorch-forecasting/blob/602ee157d18bd00ea7e1cfddeb0c8184c138976f/pytorch_forecasting/tests/test_all_v2/test_all_estimators_v2.py#L134

and relax the naming test to allow both valid patterns:

Allowed Forms:

- If estimator is present in both versions:
```
MyForecaster_v2  →  MyForecaster_pkg_v2
```
- If estimator is only present in v2:
```
MyForecaster  →  MyForecaster_pkg_v2
```

Something like this: 
```python
assert object_pkg.__name__ == object_class.__name__[:-3] + "_pkg_v2", msg 
# ignoring the "_v2" part of the object class
```
This condition should be added in addition to the current condition:
```python
assert object_pkg.__name__ == object_class.__name__ + "_pkg_v2", msg
```
If any one of these conditions satisfies then the test should pass.
Maybe like this?
```python
expected_with_version = object_class.__name__ + "_pkg_v2"

if object_class.__name__.endswith("_v2"):
    base_name = object_class.__name__[:-3]
else:
    base_name = object_class.__name__

expected_without_version = base_name + "_pkg_v2"

assert object_pkg.__name__ in {
    expected_with_version,
    expected_without_version,
}, msg
```

--- a/pytorch_forecasting/tests/test_all_v2/test_all_estimators_v2.py
+++ b/pytorch_forecasting/tests/test_all_v2/test_all_estimators_v2.py
@@ -124,6 +124,7 @@ class TestAllPtForecastersV2(EstimatorPackageConfig, EstimatorFixtureGenerator):
         )
         assert object_pkg.name() == object_class.__name__, msg
 
+        # check naming convention
         class_name = object_class.__name__
 
         expected_names = {class_name + "_pkg_v2"}

--- a/pytorch_forecasting/tests/test_all_v2/test_all_estimators_v2.py
+++ b/pytorch_forecasting/tests/test_all_v2/test_all_estimators_v2.py
@@ -124,11 +124,17 @@ class TestAllPtForecastersV2(EstimatorPackageConfig, EstimatorFixtureGenerator):
         )
         assert object_pkg.name() == object_class.__name__, msg
 
-        # check naming convention
+        class_name = object_class.__name__
+
+        expected_names = {class_name + "_pkg_v2"}
+
+        if class_name.endswith("_v2"):
+            expected_names.add(class_name[:-3] + "_pkg_v2")
+
         msg = (
-            f"Package {object_pkg.__name__} does not match class "
-            f"name {object_class.__name__}. "
-            "The expected package name is "
-            f"{object_class.__name__}_pkg."
+            f"Package class '{object_pkg.__name__}' does not follow the expected "
+            f"naming convention for estimator '{class_name}'. "
+            f"Expected one of: {sorted(expected_names)}."
         )
-        assert object_pkg.__name__ == object_class.__name__ + "_pkg_v2", msg
+
+        assert object_pkg.__name__ in expected_names, msg

